spring-web/src/main/java/org/springframework/http/client/JdkClientHttpResponse.java

```diff
@@ -41,22 +40,21 @@
  */
 class JdkClientHttpResponse implements ClientHttpResponse {
 
-	private final HttpResponse<InputStream> response;
+	private final int statusCode;
 
 	private final HttpHeaders headers;
 
 	private final InputStream body;
 
 
-	public JdkClientHttpResponse(HttpResponse<InputStream> response) {
-		this.response = response;
-		this.headers = adaptHeaders(response);
-		InputStream inputStream = response.body();
-		this.body = (inputStream != null ? inputStream : InputStream.nullInputStream());
+	public JdkClientHttpResponse(int statusCode, java.net.http.HttpHeaders headers, InputStream body) {
+        this.statusCode = statusCode;
+        this.headers = adaptHeaders(headers);
+		this.body = body != null ? body : InputStream.nullInputStream();
```

Comment:
I think we could have constructors like these:

```
public JdkClientHttpResponse(HttpResponse<InputStream> response) {
    this(response, response.getInputStream());
}

public JdkClientHttpResponse(HttpResponse<InputStream> response, InputStream body) {

}
```

That would keep the logic mostly as it was before and avoid the full qualified `java.net.http.HttpHeaders` type.